### PR TITLE
Add Bridges

### PR DESCRIPTION
**Pull request by melotik**
Questions for reviewers
- Is Amount in and amount out used correctly?

### CODE REVIEW

File: schemas/bridge/SCHEMA.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+List of pools in bridge protocols (applicable for pool and mint based bridges).
+
+| Property                | Description                                               | Type   |
+|-------------------------|-----------------------------------------------------------|--------|
+| creation_timestamp       | The timestamp this pool was created.                      | number |
+| creation_block_number    | The block number this pool was created on.                | number |
+| token_address            | The contract address of the token being bridged or pooled. | string |
+| token_index              | (Optional, only applicable to pool-based) The index of the pooled token in the smart contract. | number |
+| token_symbol             | The symbol of the token.                                  | string |
+| fee_rate                 | The fee rate as a percentage for the pool, if applicable (ie, 2.3% fee as 2.3). | number |
```
Comment (0xroll) [inline schemas/bridge/SCHEMA.md]: lets standardize the fee_rate format.

`ie, 2.3% fee as 0.023`

File: schemas/bridge/SCHEMA.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+Standard table definitions for bridge protocols, including pool-based, mint-based, and intent-based bridges.
+
+## Version: 1.0.0-alpha
+
+### Tokens
+
+List of tokens in bridge protocols.
+
+| Property                | Description                                               | Type   |
```
Comment (0xroll) [inline schemas/bridge/SCHEMA.md:11]: can we add in

token_index

File: schemas/bridge/SCHEMA.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+| timestamp                | The timestamp of the transaction.                         | number |
+| block_number             | The block number of the transaction.                      | number |
+| transaction_hash         | The hash of the transaction this bridge originated from.  | string |
+| source_chain_id          | The source standard chain ID.                             | number |
+| destination_chain_id     | The destination standard chain ID.                        | number |
+| volume_usd               | The volume of the transaction in USD.                     | number |
+| user_address             | The address of the user initiating the transaction.       | string |
+| solver_address           | The address of the solver (for intent-based bridges).     | string |
+| pool_address             | The address of the pool (for pool-based and mint-based bridges). | string |
+| bridge_type              | The type of bridge (pool-based, mint-based, or intent-based). | string |
```
Comment (0xroll) [inline schemas/bridge/SCHEMA.md]: lets add in a few more columns

`log_index`
`token_address`
`token_amount`

maybe `volume_usd` to `token_amount_usd`

Comment (0xroll) [reply schemas/bridge/SCHEMA.md]: are we doing `transaction_hash` or `transaction_id`?

File: schemas/bridge/SCHEMA.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+### Maker Snapshot
+
+List of makers in intent-based bridge protocols.
+
+| Property                | Description                                               | Type   |
+|-------------------------|-----------------------------------------------------------|--------|
+| timestamp                | The timestamp of the snapshot.                            | number |
+| user_address             | The address of the maker.                                 | string |
+| amount_in_usd            | The amount received in USD, during this snapshot.         | number |
```
Comment (0xroll) [inline schemas/bridge/SCHEMA.md:112]: token_address
token_in_amount
token_out_amount

File: schemas/bridge/SCHEMA.md [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+| amount_out_usd           | The amount sent in USD, during this given snapshot.       | number |
+| fees_usd                 | The fees collected in USD, during this snapshot.          | number |
+| chain_id                 | The standard chain ID where the maker resides.            | string |
+
+### Taker Snapshot
+
+List of takers in intent-based bridge protocols.
+
+| Property                | Description                                               | Type   |
+|-------------------------|-----------------------------------------------------------|--------|
```
Comment (0xroll) [inline schemas/bridge/SCHEMA.md:123]: token_address
token_amount

File: schemas/bridge/SCHEMA.md [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+| user_address             | The address of the taker.                                 | string |
+| volume_usd               | The volume bridged in USD, in the given snapshot.         | number |
+| chain_id                 | The standard chain ID where the taker resides.            | number |
+
+### Token Snapshot
+
+List of takers in intent-based bridge protocols.
+
+| Property                | Description                                               | Type   |
+|-------------------------|-----------------------------------------------------------|--------|
```
Comment (0xroll) [inline schemas/bridge/SCHEMA.md:136]: token_amount

File: schemas/bridge/SCHEMA.md (hunk 3 above)
Comment (melotik) [reply schemas/bridge/SCHEMA.md]: I've been using `transaction_hash` elsewhere. There is actually an update to nft/dex ag to that, will submit that.

File: schemas/bridge/SCHEMA.md (hunk 5 above)
Comment (melotik) [reply schemas/bridge/SCHEMA.md:123]: Also, I updated `volume_usd` to `token_amount_usd`. Lmk if the descriptions make sense in this case, I am less familiar with intent-based bridges

Comment (melotik) [reply schemas/bridge/SCHEMA.md:123]: I guess I am confused if this is a volume or a balance?

File: schemas/bridge/SCHEMA.md (hunk 6 above)
Comment (melotik) [reply schemas/bridge/SCHEMA.md:136]: To clarify, this only covers pool based bridges? I also added `token_amount_usd`

File: schemas/bridge/SCHEMA.md (hunk 5 above)
Comment (0xroll) [reply schemas/bridge/SCHEMA.md:123]: this should be volume. to get the volume in token amount

File: schemas/bridge/SCHEMA.md (hunk 6 above)
Comment (0xroll) [reply schemas/bridge/SCHEMA.md:136]: not that familiar with all type of bridges , but the goal here is to ensure we have the token amount column for all bridges

File: schemas/bridge/SCHEMA.md (hunk 5 above)
Comment (melotik) [reply schemas/bridge/SCHEMA.md:123]: Okay, then I am using `volume_amount` and `volume_usd`

File: schemas/bridge/SCHEMA.md (hunk 6 above)
Comment (melotik) [reply schemas/bridge/SCHEMA.md:136]: Then I have added token amount and volume amounts and their corresponding USD fields.

- I belv you could have token amounts (or supplied amounts in pools), for pool based bridges
- All bridges have a volume bridged amount

**Review by 0xroll — APPROVED** [review]

**Pull request merged**